### ISSUE TITLE
Pie chart how to break line in dataLabels?

### ISSUE THREAD

**Issue by nlpsl202** [apexcharts/apexcharts.js#670]
I want to let my dataLabels break line,how do i do?
now:
![A](https://user-images.githubusercontent.com/8954021/59328627-6c2e7f80-8d1f-11e9-8c49-b3a0760e2255.PNG)

I wnat this:
![image](https://user-images.githubusercontent.com/8954021/59328667-8b2d1180-8d1f-11e9-819f-ade2a5dc8bc0.png)

I have try \n,\r\n,<br/> but not work.
my code now is:
`dataLabels: {
                    enabled: true,
                    style: {
                        fontSize: '15px'
                    },
                    formatter: function (value, opts) {
                        return [opts.w.globals.labels[opts.seriesIndex], value + '%']
                    }
                },`

**Comment by Demak** [comment]
There seems to be no other option to do that other than modifying chart yourself, I have been working on similar functionality and here is a solution modified for your use case:
`function break_lines(chartContext, config){
	chartContext.w.globals.dom.baseEl.querySelectorAll('.apexcharts-pie-label tspan').forEach((el)=>{
		var index = Array.prototype.indexOf.call(el.parentNode.childNodes, el);
		el.setAttribute('x', el.parentNode.getAttribute('x'));
		if(index) el.setAttribute('dy', '1.2em');
	});
}`
This function will adjust the tspan's inside labels below each other. It should be called after each chart redraw. The most relevant events for that are 'mounted' and 'updated'.
`chart: {
	...
	type: 'pie',
	events: {
		mounted: break_lines,
		updated: break_lines,
	}
},`
Unfortunately 'updated' event seems to be raised not immediately after redraw, so labels will appear on a single line for a moment when browser window is resized.

**Comment by ws420** [comment]
I

> There seems to be no other option to do that other than modifying chart yourself, I have been working on similar functionality and here is a solution modified for your use case:
> 
> function break_lines(chartContext, config){
> 	chartContext.w.globals.dom.baseEl.querySelectorAll('.apexcharts-pie-label tspan').forEach((el)=>{
> 		var index = Array.prototype.indexOf.call(el.parentNode.childNodes, el);
> 		el.setAttribute('x', el.parentNode.getAttribute('x'));
> 		if(index) el.setAttribute('dy', '1.2em');
> 	});
> }
> This function will adjust the tspan's inside labels below each other. It should be called after each chart redraw. The most relevant events for that are 'mounted' and 'updated'.
> 
> chart: {
> 	...
> 	type: 'pie',
> 	events: {
> 		mounted: break_lines,
> 		updated: break_lines,
> 	}
> },
> Unfortunately 'updated' event seems to be raised not immediately after redraw, so labels will appear on a single line for a moment when browser window is resized.

I tried this but my pie chart looks like this:
![image](https://user-images.githubusercontent.com/48076840/69471826-c40eef00-0d71-11ea-9803-f1cd53782302.png)

Any idea why?

**Comment by junedchhipa** [comment]
Since v3.15.2, multiline labels can be achieved by providing an array of text in the `formatter` property of dataLabales.

```javascript
dataLabels: {
  formatter(val, opts) {
    const name = opts.w.globals.labels[opts.seriesIndex]
    return [name, val.toFixed(1) + '%']
  }
}
```

Full sample code - https://github.com/apexcharts/apexcharts.js/blob/46a8bd5bb4e8227d0f0c55e950b783d7ce05f9b7/samples/vanilla-js/pie/monochrome-pie.html

![Screen Shot 2020-01-24 at 12 12 09 PM](https://user-images.githubusercontent.com/17950663/73049519-da851400-3ea2-11ea-8ba7-9948ce88f284.png)

**Comment by sasipannon** [comment]
Hi,
I using 3.22.2 and try this, but still get error, cause in "apex-types.d.ts" formatter is still this:
line 708: 
`formatter?(val:` string | number | number[], opts?: any): string | number;`

Thanks for your help!

**Comment by MuhammadBilal-arch** [comment]
Hello I have same issue

`    dataLabels: {
      enabled: true,
      style: {
        fontSize: '16px',
        fontWeight: 'medium', 
        colors: ['#7F56D9', '#F4EBFF'],      
      },
      
      textAnchor: 'middle',
      formatter: function (val, { seriesIndex }) {
        return seriesIndex === 0 ? `$${state?.series[0]}` : `$${state?.series[1]}`;
      },
    },`
    
    colors are not working it show gray color 
    
![image](https://github.com/apexcharts/apexcharts.js/assets/67434155/7fda6e62-b16c-4198-86c7-e5ab70ee6ac8)

anyway to fix this issue? I'm using "apexcharts": "^3.37.0",
